packages/staking/src/features/drawer/StakePoolPreferences.tsx

```diff
@@ -47,12 +48,15 @@ export const StakePoolPreferences = () => {
     setIsDrawerVisible: store.setIsDrawerVisible,
   }));
 
-  const displayData = draftPortfolio.map(({ name, id }, i) => ({
+  const displayData = draftPortfolio.map(({ name = '-', weight, id }, i) => ({
     color: PIE_CHART_DEFAULT_COLOR_SET[i] as PieChartColor,
     id,
-    name: name || '',
-    value: 1,
+    name,
+    weight,
   }));
+  const createRemovePoolFromPortfolio = (poolId: Wallet.Cardano.PoolIdHex) => () => {
```

Comment:
```suggestion
  const handleRemovePoolFromPortfolio = (poolId: Wallet.Cardano.PoolIdHex) => () => {
```
Personally I liked the original name more. Create and Remove in succession might disturb the reader

Reply:
I understand but it's not a handler yet. You need to create it dynamically, passing `poolId` so we can rename it to `createHandleRemovePoolFromPortfolio` if you wish 😄